ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -130,12 +133,23 @@ public class RNTrackPlayer: RCTEventEmitter, AudioPlayerDelegate {
     
     @objc(setupPlayer:resolver:rejecter:)
     public func setupPlayer(config: [String: Any], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
-        do {
-            try AVAudioSession.sharedInstance().setCategory(.playback, mode: .default)
-            try AVAudioSession.sharedInstance().setActive(true)
-        } catch {
-            reject("setup_audio_session_failed", "Failed to setup audio session", error)
+
+        //configure base category -- defaults to .playback
+        let sessionCategory: String = config["iosCategory"] as? String ?? "playback"
+        self.sessionCategory = SessionCategory(rawValue: sessionCategory)!.mapConfigToAVAudioSessionCategory()
```

Comment:
This has the bug that if you give it a `iosCategory` but it's not a valid one, it will crash because of the force unwrap (!).

It'd be safer to write this as:
```swift
if let sessionCategory = config["iosCategory"] as? String {
  let mappedCategory = SessionCategory(rawValue: sessionCategory)
  self.sessionCategory = (mappedCategory ?? .playback).mapConfigToAVAudioSessionCategory()
}
```